controllers/azurecluster_controller.go

```diff
@@ -263,6 +263,7 @@ func (acr *AzureClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	return reconcile.Result{}, nil
 }
 
+//nolint:unparam // Always returns an empty struct for reconcile.Result
```

Comment:
The `unparam` linter is complaining here (and elsewhere) that all code paths through the function return `reconcile.Result{}` for the first parameter, suggesting it should be removed from the function signature altogether.

I had the linter ignore it because that refactoring seemed out of scope, but I can pursue that if desired.

Reply:
Agree refactoring this is out of scope here. I think keeping the signature similar to `reconcileNormal` and `reconcileDelete` helps reinforce that this is a logical sibling to those anyway.

Reply:
Agreed. We could consider making an actual `interface` here too, which is another way to silence `unparam`.